cmd/start.go

```diff
@@ -127,7 +128,29 @@ func Start(c *cli.Context) error {
 
 	nodeHeader(networkProvider.ConnectionManager().AddrStrings(), config.LibP2P.Port)
 
-	handle, err := persistence.NewDiskHandle(config.Storage.DataDir)
+	// Below, use the bare data dir for the Ethereum chain for backwards
+	// compatibility. A future version may do a one-time migration of the
+	// ethereum directory.
+	//
+	// For other chains, lightly sanitize the chain's self-reported name and use
+	// it as a path prefix within the data directory. Since all directories in
+	// the Ethereum path are Ethereum addresses, the sanitize pass drops any 0x
+	// in the chain name to avoid clashes.
+	diskPersistencePath := config.Storage.DataDir
+	if chainHandle.Name() != "ethereum" {
+		diskPersistencePath += "/" +
+			// TODO allowlist vs sanitization? [A-Za-z0-9-_.] and replace all
+			//      else with -?
+			// TODO max length
+			strings.NewReplacer(
+				"/", "", // no path delimiters please
+				" ", "-", // spaces become dashes
+				"0x", "", // no 0x to avoid address clashes
+			).Replace(
+				strings.ToLower(chainHandle.Name()),
+			)
```

Comment:
Should we extract this part to some utility function in... ... Well, I do not know where would that fit into, heh. We are clearly reformatting some chain-specific string into something fitting our expected directory layout.

Reply:
I'm going to rework `chainHandle.Name()` to require this behavior in the first place in the docs, and then going to impose a failing check for it here I think.

Reply:
This is redone now, once I fix the commits it'll make a little more sense hehe.